### PR TITLE
[WIP] Stream searches use our search API

### PR DESCRIPTION
**Pull request by gergely-ujvari**
Stream search will no longer use websocket to transfer data to client, instead it will call our regular search API.
"any" field is handled by api.py, it creates a multi-match query for it.

**Lost functionality**
- The since search keyword is no longer working. For that to work we have to further enhance our search API

What's left:
- Improve code-readability
- Decide whether this is good enough for us.

### CODE REVIEW

File: h/static/scripts/streamsearch.coffee [hunk 1]
```diff
@@ -30,8 +27,6 @@ class StreamSearchController
 
     $scope.shouldShowThread = (container) -> true
 
-    streamer.send({filter: streamfilter.getFilter()})
```
Comment (tilgovi) [inline h/static/scripts/streamsearch.coffee]: I think you still want to send this, but just change it to use past_data: none. No?

File: h/static/scripts/vendor/annotator.store.js [hunk 2]
```diff
@@ -194,7 +194,8 @@
         headers: this.element.data('annotator:headers'),
         dataType: "json",
         success: onSuccess || function() {},
-        error: this._onError
+        error: this._onError,
+        traditional: true
```
Comment (tilgovi) [inline h/static/scripts/vendor/annotator.store.js]: What's this change for? It doesn't appear to be used.

**Comment by tilgovi** [comment]
I'm not sure, but I think you may have completely over-thought this.
Elasticsearch can interpret a query strings like /_search?q=text:foo,user:bar.
There may be some simplification you can make wherein you simply pass the search bar text straight to backend, rather than converting it to an object. And maybe some simplification to use a different API of Elasticsearch to have it parse the parameters (you may need to split/re-join/urlencode) rather than all the multi_match munging.

File: h/static/scripts/streamsearch.coffee (hunk 1 above)
Comment (gergely-ujvari) [inline h/static/scripts/streamsearch.coffee]: Yes, that's right for updates. Good catch.

File: h/static/scripts/vendor/annotator.store.js (hunk 2 above)
Comment (gergely-ujvari) [inline h/static/scripts/vendor/annotator.store.js]: It is used when I call the `loadAnnotationsFromSearch` function.

**Comment by gergely-ujvari** [comment]
@tilgovi:

Question: Do we want to support a different search api format? Because our current api accepts only the `?field1=val1&field2=val2...` format.

And if yes, then the backend would still have to transform the url. I.E. from "test user:johndoe quote:the in" it has to produce something like: 
`_/search?=user:johndoe and quote:the and (user:test,in or quote:test,in or text:test,in or tags:text,in)` do you find this mungling better than the "multi_match" mungling? (And of course I do not know that the annotator-store currently supports search_uri for ES or not)

+1: The [ES page](http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-uri-request.html) says that the uri search maps to Query String, which is not always ideal for the performance. (as Query String supports wildcards, and many more things that can have performance decrease).

**Comment by tilgovi** [comment]
Good points. I don't know.

File: h/static/scripts/searchfilters.coffee [hunk 3]
```diff
@@ -43,6 +43,28 @@ class SearchFilter
...
+    if searchtext
+      terms = @_tokenize(searchtext)
+      for term in terms
+        filter = term.slice 0, term.indexOf ":"
+        unless filter? then filter = ""
+        if (term.indexOf ":") > -1
+          data = term.slice term.indexOf(":")+1
+          addToObj(filter, data)
+        else
+          addToObj('any', term)
```
Comment (aron) [inline h/static/scripts/searchfilters.coffee]:
```coffee
      for term in terms
        [filter, data] = term.split(':')
        unless data
          filter = 'any'
          data = term
        addToObj(filter, data)
```

**Comment by gergely-ujvari** [comment]
Closing in favor of #1862

**Pull request closed without merging**